### PR TITLE
Add maintainers.md

### LINKED ISSUE #589: [DOC-Meta] Add a MAINTAINERS.md
For an example MAINTAINERS.md see [the one from the .github](https://github.com/opensearch-project/.github/blob/main/MAINTAINERS.md) project.

We should have clear documentation who is maintaining this repository.

### PR DESCRIPTION
**Pull request by carolxob**
Signed-off-by: carolxob <carolxob@amazon.com>

### Description
Added new maintainers.md file content.

### Issues Resolved

Closes #589 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Naarcha-AWS — APPROVED** [review]

**Review by dblock — COMMENTED** [review]
Add the bottom section from https://github.com/opensearch-project/OpenSearch/blob/main/MAINTAINERS.md about what maintainers do and how to contribute.

**Review by hdhalter — COMMENTED** [review]
After adding DB's input, this is good to go. We got the sign-off from Marketing.

**Review by hdhalter — APPROVED** [review]

**Pull request merged**
